ledger/acctupdates.go

```diff
@@ -1478,14 +1481,14 @@ func (au *accountUpdates) lookupImpl(rnd basics.Round, addr basics.Address, with
 			withRewards = false
 		}
 
-		// Check if this is the most recent round, in which case, we can
-		// use a cache of the most recent account state.
-		if offset == uint64(len(au.deltas)) {
-			macct, ok := au.accounts[addr]
-			if ok {
+		// check if we've had this address modified in the past rounds. ( i.e. if it's in the deltas )
+		macct, indeltas := au.accounts[addr]
+		if indeltas {
+			// Check if this is the most recent round, in which case, we can
+			// use a cache of the most recent account state.
+			if offset == uint64(len(au.deltas)) {
 				return macct.data, nil
 			}
-		} else {
 			// Check if the account has been updated recently.  Traverse the deltas
```

Comment:
The updated comment in lookupWithoutRewardsImpl is very nice. Will be good to copy it here as well. 
			// the account appears in the deltas, but we don't know if it appears in the
			// delta range of [0..offset], so we'll need to check :
			// Traverse the deltas backwards to ensure that later updates take
			// priority if present.